simpl/rest.py

```diff
@@ -53,3 +57,142 @@ def wrapped(*args):
             return fxn(data, *args)
         return wrapped
     return wrap
+
+
+def write_body(data, request, response):
+    """Write output with format based on accept header.
+
+    This cycles through the global HANDLERs to match content type and then
+    calls that handler. Additional handlers can be added to support Additional
+    content types.
+    """
+    accept = request.get_header('Accept', ['application/json'])
+
+    for content_type in HANDLERS:
+        if content_type in accept:
+            return HANDLERS[content_type](data, request, response)
+
+    # Use default
+    return HANDLERS['default'](data, request, response)
+
+
+def write_json(data, request, response):
+    """Write output in json."""
+    response.set_header('Content-Type', 'application/json')
+    return to_json(data)
+
+
+def to_json(data):
+    """Write out python object to JSON.
+
+    Includes special handling for custom objects derived from MutableMapping.
+    """
+    if isinstance(data, collections.MutableMapping) and hasattr(data, 'dumps'):
+        return data.dumps(indent=4)
+    return json.dumps(data, indent=4)
+
+
+HANDLERS = {
+    'application/json': write_json,
+    'default': write_json
+}
```

Comment:
What do you think about implementing something generic (multiple handlers) once we really need it and we do json only for now? Might be able to simplify this PR.